include/openthread/platform/radio.h

```diff
@@ -505,6 +505,37 @@ void otPlatRadioSetExtendedAddress(otInstance *aInstance, const otExtAddress *aE
  */
 void otPlatRadioSetShortAddress(otInstance *aInstance, otShortAddress aShortAddress);
 
+/**
+ * Get current PAN interface id.
+ *
+ * @param[in]  aInstance      The OpenThread instance structure.
+ * @param[out] aInterfaceId   Pointer to the variable for storing interface id
+
+ * @retval  OT_ERROR_NONE               Successfully got the property.
+ * @retval  OT_ERROR_NOT_IMPLEMENTED    Failed due lack of the support in radio or platform supprts
+ *                                      all interfaces simultaneously
+ *                                      (i.e. no active/inactive interface concept in the platform level)
+ *
+ */
+otError otPlatGetMultipanInterface(otInstance *aInstance, uint8_t *aInterfaceId);
+
+/**
+ * Select PAN interface id to use for communication.
+ *
+ * @param[in] aInstance         The OpenThread instance structure.
+ * @param[in] aInterfaceId      Id of the interface to select.
+ * @param[in] aCompletePending  True if ongoing radio operation should complete before interface switch (Soft switch).
+ *
+ * @retval  OT_ERROR_NONE               Successfully set the property.
+ * @retval  OT_ERROR_BUSY               Failed due to another operation on going.
+ * @retval  OT_ERROR_NOT_IMPLEMENTED    Failed due to lack of support in radio for the given interface id or
+ *                                      platform supprts all interfaces simultaneously
+ *                                      (i.e. no active/inactive interface concept in the platform level)
+ * @retval  OT_ERROR_ALREADY            Given interface is already active.
+ *
+ */
+otError otPlatSetMultipanInterface(otInstance *aInstance, uint8_t aInterfaceId, bool aCompletePending);
```

Comment:
This function probably requires better description. Maybe also argument name should be changed to something like aRadioInterface.
It is not intended to map IID to instance. InterfaceId is not actually IID number, it is radio interface id. But for easier development it is best to keep InterfaceId equal to IID. There may be less radio interfaces than there is IID's.
This function is intended only for platforms that do not support parallel communication on both interfaces, for example use a single HW radio and two PANs operate on different channels. This API sets currently active interface in such cases as the communication can not happen in parallel in such cases. If all interfaces are in receive mode one that was specified by otPlatSetMultipanInterface will be actually receiving.
For platforms that can handle communication in parallel this function will do nothing.

I will correct the description of this function.

Reply:
Thanks @MarekPorwisz.

Can we directly use the `otInstance *` itself in these APIs to indicate the PAN that need to take control of the single/common radio? 

Something like:
```c++
// Set `aInstance` as the current active instance controlling radio
otError otPlatMultipanSetActiveInstance(otInstance *aInstance); 
``` 
I think the main idea is to use different `otInstance` objects to represent different PANs.

Do we need to have the notion of an `uint8_t` "radio interface ID" in these APIs?

Reply:
I've been thinking more on  this, and I understand why these APIs in `radio.h` are needed/added. However, I'm not sure if `radio.h` is the best place for them.

Here are my thoughts:

- We want to use OT's "multiple instance" model, where each `otInstance` corresponds to a different PAN.
- Each `ot::Instance` will assume that it has an `ot::Radio` and can control it using the `otPlatRadio` APIs.
- We want the platform layer to handle multi-PAN logic.
- This is mainly for the case where there is one radio and we want to share it between PANs. In this case, the platform layer needs to make it look like there are multiple radios, each controlled by a different set of `otPlatRadio()` APIs called with a different `otInstance *` input.
- I think the `ot::Radio` module should not be aware of the multi-PAN feature. Therefore, I'm not sure about adding these APIs to `radio.h` and related modules.
- One solution would be to define a new `openthread/platform/multipan.h` platform header and define these APIs in that header. 
    - This would give us a dedicated place to define APIs for multi-PAN support, and it would also allow us to add more APIs in the future if needed (e.g., APIs for arbitrating conflicting requests).

What do you think?

Reply:
OK, I will prepare a separate API file for multipan and update the API to use Instance instead of radio interface id.
Please note that this changed otPlatMultipanSetActiveInstance can only be implemented on the RCP side. Host side will need to use RadioSpinel and specify IID of the active interface. This is because there is not mapping of the instances to IID on the host side and instances are not aware of each other existence. I even assume it is possible to implement every instance as a separate process, where there is no access to the other instance objects.

Reply:
Thanks. 

The way I see this is we want to use "multiple instance" feature to realize multi-PAN.  Each `ot::Instance` (whether on host or on RCP) runs on its own and gets to call all platform APIs, including `otPlatRadio()`, using its `otInstance *`. Each `ot::Instance` assumes that it is sole controller of all its platform/radio.

To support multiple-PAN when there is really a single radio which we want to share, each PAN is running its own `otInstance` and is not aware of existence of others. We would need a module/process that arbitrate the action of single radio (make it look like and behave as if it is multiple radios). This module is aware of all `otInstance` objects (each corresponding to a PAN) and can make decisions based on the plat API commands issued to radio by each `otInstance` (e.g., which instance is allowed to actively using/controlling radio at a time). 
This module itself would not be tied to any particular `ot::Instance` and can use the `otPlatMultipan{}()` APIs to control the radio (which are special platform APIs in the sense that they are also not tied to any `otInstance *`).

Does this make make sense?

Reply:
I've added additional commit that separates multipan API. On the RCP side Instance object is used for selection of the interface. On the host side IID is used. I don't see any pretty way to select active interface using instance on the host side as the instances do not need to belong to the same process nor it needs to be an actual instance object e.g if another interface is not Thread. The uart multiplexer could be implemented on the HDLC level using some IPC mechanisms, and the arbiter probably on the host application side. The secondary protocol could request switching interfaces for a short period of time, then return it to the main after communication is complete (e.g SED/SSED). 
Until there is any actual implementation that makes a use of this we won't know what API will suits best. We should assume we will be changing it in the future.

Ave also created some tests for the switching api in this commit.